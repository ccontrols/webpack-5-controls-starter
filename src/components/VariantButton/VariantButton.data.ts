export default {
  overview: {
    "1": {
      text: "Cecelia Langosh",
      variant: "success",
      icon: "none"
    },
    "2": {
      text: "Jolie Beahan Sr.",
      variant: "disabled",
      icon: "down-arrow"
    },
    "3": {
      text: "Kane Ortiz",
      variant: "warning"
    },
    "4": {
      text: "Ronny Spinka",
      variant: "success",
      icon: "right-arrow"
    },
    "5": {
      text: "Jayce Schaden",
      icon: "up-arrow"
    }
  }
};
