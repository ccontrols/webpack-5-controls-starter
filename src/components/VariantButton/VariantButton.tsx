import React, { FC } from 'react';

type ButtonVariant =
  | 'primary'
  | 'accent'
  | 'disabled'
  | 'success'
  | 'warning'
  | 'error';

const variant_colors: Record<ButtonVariant, string> = {
  primary: '#F2F2F2',
  accent: '#F2F2F2',
  disabled: '#828282',
  success: '#F2F2F2',
  warning: '#4F4F4F',
  error: '#F2F2F2',
};

const variant_backgrounds: Record<ButtonVariant, string> = {
  primary: '#2F80ED',
  accent: '#9B51E0',
  disabled: '#E0E0E0',
  success: '#219653',
  warning: '#F2C94C',
  error: '#EB5757',
};

export interface VariantButtonProps {
  /**
   *  button label text
   */
  text: string;

  /**
   *  variant kind
   */
  variant?: ButtonVariant;
}

/**
 * Button with variants
 */
export const VariantButton: FC<VariantButtonProps> = ({
  text,
  variant = 'primary',
}) => {
  return (
    <button
      disabled={variant === 'disabled' ? true : undefined}
      style={{
        color: variant_colors[variant],
        backgroundColor: variant_backgrounds[variant],
        borderRadius: 8,
        border: `1px solid #000000`,
        display: 'flex',
        alignItems: 'center',
      }}
    >
      <div data-testid="label" style={{ padding: '5px 10px' }}>
        {text}
      </div>
    </button>
  );
};
