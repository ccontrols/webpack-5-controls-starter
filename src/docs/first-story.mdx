---
title: First Story
---
import { Button } from '../components/Button';
import { Playground, Story, PropsTable,  ComponentSource  } from '@component-controls/blocks';

# My first doc story

## Component source
<ComponentSource of="." title='Component source' />

## Playground
<Playground description="Button story">
  <Story name="simple" component={Button}>
    {({ backgroundColor }) => <Button backgroundColor={backgroundColor}>click me</Button>}
  </Story>
</Playground>

## Component Props
<PropsTable of="." visibility="controls"/>