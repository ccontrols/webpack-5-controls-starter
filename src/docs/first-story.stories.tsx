import React from 'react';
import { Button, ButtonProps } from '../components/Button';

export default {
  title: 'ES Story',
  component: Button,
}

export const overview = (props: ButtonProps) => <Button {...props}>click me</Button>;

