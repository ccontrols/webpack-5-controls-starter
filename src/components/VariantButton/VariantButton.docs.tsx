import React from 'react';
import { Document, Example } from '@component-controls/core';
import { VariantButton, VariantButtonProps } from './VariantButton';

export default {
  title: 'VariantButton',
  component: VariantButton,
} as Document;

export const overview: Example<VariantButtonProps> = props => (
  <VariantButton {...props} />
);

overview.controls = {
  text: 'Button',
};
