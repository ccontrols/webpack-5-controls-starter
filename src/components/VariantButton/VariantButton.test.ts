import * as path from "path";
import { run, AxeResults } from "axe-core";
import { reactRunDOM } from "@component-controls/test-renderers";
import "@component-controls/jest-axe-matcher";
import { loadConfigurations } from "@component-controls/config";
import { renderDocument } from "@component-controls/test-renderers";
import { render, act } from "@testing-library/react";
import { renderErr } from "@component-controls/test-renderers";

import * as examples from "./VariantButton.docs";
import data from "./VariantButton.data";

describe("VariantButton", () => {
  const configPath = path.resolve(__dirname, "../../../.config");
  const config = loadConfigurations(configPath);
  let renderedExamples: ReturnType<typeof renderDocument> = [];
  act(() => {
    renderedExamples = renderDocument(examples, config, data);
  });
  if (!renderedExamples) {
    renderErr();
    return;
  }
  renderedExamples.forEach(({ name, rendered, dataId, values }) => {
    describe(name, () => {
      const runTests = (): void => {
        it("snapshot", () => {
          const { asFragment } = render(rendered);
          expect(asFragment()).toMatchSnapshot();
        });
        it("accessibility", async () => {
          const axeResults = await reactRunDOM<AxeResults>(rendered, run);
          expect(axeResults).toHaveNoAxeViolations();
        });
      };
      if (dataId && values) {
        describe(dataId, runTests);
      } else {
        runTests();
      }
    });
  });
});
