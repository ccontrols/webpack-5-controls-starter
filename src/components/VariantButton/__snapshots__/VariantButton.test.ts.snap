// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VariantButton overview 1 snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: rgb(242, 242, 242); background-color: rgb(47, 128, 237); border-radius: 8px; border: 1px solid #000000; display: flex; align-items: center;"
  >
    <div
      data-testid="label"
      style="padding: 5px 10px;"
    >
      Cecelia Langosh
    </div>
  </button>
</DocumentFragment>
`;

exports[`VariantButton overview 2 snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: rgb(242, 242, 242); background-color: rgb(47, 128, 237); border-radius: 8px; border: 1px solid #000000; display: flex; align-items: center;"
  >
    <div
      data-testid="label"
      style="padding: 5px 10px;"
    >
      Jolie Beahan Sr.
    </div>
  </button>
</DocumentFragment>
`;

exports[`VariantButton overview 3 snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: rgb(242, 242, 242); background-color: rgb(47, 128, 237); border-radius: 8px; border: 1px solid #000000; display: flex; align-items: center;"
  >
    <div
      data-testid="label"
      style="padding: 5px 10px;"
    >
      Kane Ortiz
    </div>
  </button>
</DocumentFragment>
`;

exports[`VariantButton overview 4 snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: rgb(242, 242, 242); background-color: rgb(47, 128, 237); border-radius: 8px; border: 1px solid #000000; display: flex; align-items: center;"
  >
    <div
      data-testid="label"
      style="padding: 5px 10px;"
    >
      Ronny Spinka
    </div>
  </button>
</DocumentFragment>
`;

exports[`VariantButton overview 5 snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: rgb(242, 242, 242); background-color: rgb(47, 128, 237); border-radius: 8px; border: 1px solid #000000; display: flex; align-items: center;"
  >
    <div
      data-testid="label"
      style="padding: 5px 10px;"
    >
      Jayce Schaden
    </div>
  </button>
</DocumentFragment>
`;
